react-components/src/tools/requests/NewToolRequestForm.js

```diff
@@ -0,0 +1,177 @@
+import intlData from "../../tools/messages";
+import {
+    DEDialogHeader,
+    FormMultilineTextField,
+    FormTextField,
+    getMessage,
+    withI18N,
+} from "@cyverse-de/ui-lib";
+import Dialog from "@material-ui/core/Dialog";
+import DialogContent from "@material-ui/core/DialogContent";
+import Button from "@material-ui/core/Button";
+
+/**
+ *
+ * A form to collect information about users tool requests.
+ *
+ */
+class NewToolRequestForm extends Component {
+    constructor(props) {
+        super(props);
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleClose = this.handleClose.bind(this);
+        this.validateUrl = this.validateUrl.bind(this);
+    }
+
+    handleSubmit(values, actions) {
+        //test_data_path and cmd_line are required by service. Most of the times those info. can be
+        // found in the source repo. So copy the source url to those fields if not provided by user.
+        if (!values.test_data_path) {
+            values.test_data_path = values.source_url;
+        }
+        if (!values.cmd_line) {
+            values.cmd_line = values.source_url;
+        }
+        actions.setSubmitting(true);
+        this.props.presenter.submitRequest(
+            values,
+            () => {
+                actions.setSubmitting(false);
+                this.handleClose();
+            },
+            (httpStatusCode, errorMessage) => {
+                actions.setSubmitting(false);
+                actions.setStatus({ success: false, errorMessage });
+            }
+        );
+    }
+
+    handleClose() {
+        this.props.presenter.onToolRequestDialogClose();
+    }
+
+    validateUrl(value) {
+        let error;
+        if (!constants.URL_REGEX.test(value)) {
+            error = getMessage("validationErrMsgURL");
+        }
+        return error;
+    }
+
+    render() {
+        const { intl, dialogOpen } = this.props;
+        return (
+            <Dialog open={dialogOpen} disableBackdropClick disableEscapeKeyDown>
+                <DEDialogHeader
+                    heading={intl.formatMessage({
+                        id: "newToolRequestDialogHeading",
+                    })}
+                    onClose={this.handleClose}
+                />
+                <DialogContent>
+                    <Formik
+                        enableReinitialize={true}
+                        onSubmit={this.handleSubmit}
+                        render={({ errors, status, touched, isSubmitting }) => (
+                            <Form>
+                                <Field
+                                    name="name"
+                                    label={getMessage("toolNameLabel")}
+                                    required={true}
+                                    margin="dense"
+                                    component={FormTextField}
+                                />
+                                <ErrorMessage name="name" component="div" />
```

Comment:
🤔 When I've used Formik, I've not needed to include the `ErrorMessage` component because most of the `Form*` components display an error for you. Are we trying to include an error if the field is marked as required?  We might be able to update all the `Form*` components to factor that in (or at least FormTextField).

Reply:
Not sure. Since I was new to Formik, I was following this tutorial - https://jaredpalmer.com/formik/docs/tutorial

Reply:
I removed the ErrorMessage component and it worked fine. Attached a screenshot.